packages/nvidia_nat_core/tests/nat/front_ends/fastapi/test_step_adaptor.py

```
@@ -525,3 +665,213 @@ def test_function_events_in_custom_mode(step_adaptor_custom, make_intermediate_s
     # Steps should still be added to history
     assert step_adaptor_custom._history[-2] is step_start
     assert step_adaptor_custom._history[-1] is step_end
+
+
+# --------------------
+# Tests for thought_description override
+# --------------------
+def test_function_start_with_thought_description_override(step_adaptor_default, make_intermediate_step):
```

Ruff (0.15.5)

[warning] Unused function argument: `make_intermediate_step`

(ARG001)